state/modelmigration.go

```diff
@@ -856,12 +856,33 @@ func (st *State) LatestRemovedModelMigration() (ModelMigration, error) {
 	return mig, nil
 }
 
+// LatestRemovedModelMigrationByUUID returns the most recent ModelMigration (if any)
+// for a model that has been removed from the state after a successful
+// migration to another controller given by the modelUUID
+func (st *State) LatestRemovedModelMigrationByUUID(modelUUID string) (ModelMigration, error) {
```

Comment:
I would rename the method above this to `CompleteMigration` to cut down on work-sauce. It should also call this one.

Then:
```go
// CompletedMigrationForModel returns the most recent migration for the 
// input model UUID if it reached the DONE phase and caused the model 
// to be relocated.
func (st *State) CompletedMigrationForModel(modelUUID string) (ModelMigration, error) {
```